docs/preview/features/endpoint-validation.md

```diff
@@ -3,53 +3,101 @@ title: "Endpoint validation"
+using Arcus.EventGrid.WebApi.Security;
+using Microsoft.Extensions.DependencyInjection;
+
+public class Startup
 {
-    return Ok();
+    public void ConfigureService(IServiceCollection services)
+    {
+        // Default set of authorization options.
+        var options = new EventGridAuthorizationOptions();
+
+        // Looks for the 'x-api-key' header in the HTTP request and tries to match it with the secret retrieved in the secret store with the name 'MySecret'.
+        services.AddMvc(options => options.Filters.Add(new EventGridAuthorizationFilter(HttpRequestProperty.Header, "x-api-key", "MySecret", options)));
+    }
 }
 ```
 
-### Configuring Authorization with attributes
-There are 2 attributes available to secure operations:    
-- The `EventGridAuthorization` has a hard-coded secret key and is only advised for demonstration or testing purposes.  
-- The `DynamicEventGridAuthorizationAttribute` allows developer to specify a static `Func` with name `RetrieveAuthenticationSecret` to implement custom logic to retrieve the actual secret key (for example from Azure KeyVault, web.config or appsettings.json)
+For this setup to work, an Arcus secret store is required as the provided secret name (in this case `"MySecret"`) will be looked up.
+See [our offical documentation](https://security.arcus-azure.net/features/secret-store/) for more information about setting this up.
 
-These attributes will only allow an operation to be called, if the configured secret key value is found in the HTTP header or the Query String with the configured key name.
+#### Configuration
 
-#### The EventGridAuthorization attribute
-This attribute has the following public constructor.  This constructor sets the header or querystring name and the hard coded secret value.
+The `EventGridAuthorizationFilter` has some additional consumer-configurable options to influence the behavior of the authorization.
 
 ```csharp
-[EventGridAuthorization("keyName", "keyValue")]
-```
+using Arcus.EventGrid.WebApi.Security;
+using Microsoft.Extensions.DependencyInjection;
+
+public class Startup
+{
+    public void ConfigureService(IServiceCollection services)
+    {
+        var options = new EventGridAuthorizationOptions();
 
-#### The DynamicEventGridAuthorization attribute
-This attribute has the following public constructors.  This constructor sets the header or querystring name to the provided value (or defaults to 'x-api-key' for the default constructor).
+        // Indicates that the Azure Event Grid authorization should emit security events during the authorization of the request (default: `false`).
+        options.EmitSecurityEvents = true;
+
+        // Looks for the 'x-api-key' header in the HTTP request and tries to match it with the secret retrieved in the secret store with the name 'MySecret'.
+        services.AddMvc(options => options.Filters.Add(new EventGridAuthorizationFilter(HttpRequestProperty.Header, "x-api-key", "MySecret", options)));
+    }
+}
+
```

Comment:
the code section should be ended here, as the next header is still included in the code block

Reply:
Ah yes, missed that one. Thanks!